functions/fish_prompt.fish

```diff
@@ -846,6 +847,47 @@ function __bobthefish_prompt_rubies -S -d 'Display current Ruby information'
     echo -ns $ruby_glyph $ruby_version ' '
 end
 
+function __bobthefish_get_cargo_version -S -d 'Display the cargo version in a directory'
+    pushd $argv[1]
+    command rustc --version | cut -d ' ' -f 2
+    popd
+end
+
+function __bobthefish_prompt_rustc -S -d 'Display current cargo version if different than system'
+    [ "$theme_display_rustc" = '' ]
+    and return
+
+    type -fq rustc
+    or return
+
+    # If this value is non-zero we should show it as a segment.
+    set -l local_rustc_version
+
+    switch "$theme_display_rustc"
+        case always
+            set local_rustc_version (__bobthefish_get_cargo_version (__bobthefish_pwd))
+        case yes
+            cargo locate-project --message-format plain --quiet >/dev/null 2>/dev/null
+            and set local_rustc_version (__bobthefish_get_cargo_version (__bobthefish_pwd))
+        case diff
```

Comment:
This is the default behavior for most segments (e.g. ruby). Can we make two options, always and yes, and have yes do what diff currently does?

(Alternatively, if it's not to expensive to default on, it would be off/always).

Reply:
It takes 70msec to run `rustc --version`. I _think_ it's too much to run with every prompt; if you had 2-3 of these it would be a third of a second and it's noticeable. But I'll defer the final decision to you.

I've done the first part.